src/Pipit/Classes/Data/AbstractPageableDataBaseRepository.php

```diff
@@ -63,10 +71,10 @@ protected function getNewResultsPage($page,$resultsPerPage,$query,$resultsCount,
     protected function countGet() {
         $sql = "SELECT COUNT(*) {$this->getBaseQuery()}";
```

Comment:
If it works, I'd rather we just alias the count(*) to something specific than evaluate the db type again.
```suggestion
        $sql = "SELECT COUNT(*) as total {$this->getBaseQuery()}";
        ...
        return intval(current($result)['total']);
```

executeQuery returns false on anything other than success, so with the alias in place, if $result isn't false, then it will have an array with the alias as key. I don't think the additional changes involving $current are needed if we use an explicit alias in the query.

Reply:
Oh good catch. I didn't look deep enough to notice the `current()` call.